src/os_esp32.cc

```diff
@@ -748,16 +748,26 @@ class HeapSummaryCollector {
       if (i == TOIT_HEAP_MALLOC_TAG) {
         for (int j = 0; j < MAX_PROCESSES; j++) {
           if (processes_[j]) {
-            printf("  │   %7d │   %6d │    process %p │\n",
+            const uint8* uuid = processes_[j]->program()->id();
+            char uuid_buffer[37];
+            sprintf(uuid_buffer, "%08x-%04x-%04x-%04x-%04x%08x",
+                static_cast<int>(Utils::read_unaligned_uint32_be(uuid)),
+                static_cast<int>(Utils::read_unaligned_uint16_be(uuid + 4)),
+                static_cast<int>(Utils::read_unaligned_uint16_be(uuid + 6)),
+                static_cast<int>(Utils::read_unaligned_uint16_be(uuid + 8)),
+                static_cast<int>(Utils::read_unaligned_uint16_be(uuid + 10)),
+                static_cast<int>(Utils::read_unaligned_uint32_be(uuid + 12)));
+            printf("  │   %7d │   %6d │    process%4d %s │\n",
```

Comment:
Could you mark the current process and the system process somehow here? They are a bit special.